plugins/orchestrator-backend/src/service/router.ts

```diff
@@ -455,77 +454,59 @@ function setupInternalRoutes(
       if (decision.result === AuthorizeResult.DENY) {
         manageDenyAuthorization(endpointName, endpoint, _req);
       }
-      await routerApi.v2
+      return routerApi.v2
         .getWorkflowStatuses()
         .then(result => res.status(200).json(result))
-        .catch((error: { message: string }) => {
+        .catch(error => {
           auditLogRequestError(error, endpointName, endpoint, _req);
-          res
-            .status(500)
-            .json({ message: error.message || INTERNAL_SERVER_ERROR_MESSAGE });
+          next(error);
         });
     },
   );
 
   // v2
   routerApi.openApiBackend.register(
     'getWorkflowInputSchemaById',
-    async (c, req: express.Request, res: express.Response) => {
+    async (c, req: express.Request, res: express.Response, next) => {
       const workflowId = c.request.params.workflowId as string;
       const instanceId = c.request.query.instanceId as string;
       const endpointName = 'getWorkflowInputSchemaById';
       const endpoint = `/v2/workflows/${workflowId}/inputSchema`;
-
-      auditLogger.auditLog({
-        eventName: endpointName,
-        stage: 'start',
-        status: 'succeeded',
-        level: 'debug',
-        request: req,
-        message: `Received request to '${endpoint}' endpoint`,
-      });
-      const decision = await authorize(
-        req,
-        orchestratorWorkflowInstanceReadPermission,
-        permissions,
-        httpAuth,
-      );
-      if (decision.result === AuthorizeResult.DENY) {
-        manageDenyAuthorization(endpointName, endpoint, req);
-      }
-
       try {
+        auditLogger.auditLog({
+          eventName: endpointName,
+          stage: 'start',
+          status: 'succeeded',
+          level: 'debug',
+          request: req,
+          message: `Received request to '${endpoint}' endpoint`,
+        });
+        const decision = await authorize(
+          req,
+          orchestratorWorkflowInstanceReadPermission,
+          permissions,
+          httpAuth,
+        );
+        if (decision.result === AuthorizeResult.DENY) {
+          manageDenyAuthorization(endpointName, endpoint, req);
+        }
+
         const workflowDefinition =
           await services.orchestratorService.fetchWorkflowInfo({
             definitionId: workflowId,
             cacheHandler: 'throw',
           });
 
         if (!workflowDefinition) {
-          auditLogRequestError(
-            new Error(`Couldn't fetch workflow definition ${workflowId}`),
-            endpointName,
-            endpoint,
-            req,
```

Comment:
why removing this?

Reply:
the error is handled in the catch. this should just throw an error. Otherwise you have all this code duplication. We want the error handling to be consistent in one place